<launch>
    <arg name="rqt" default="false"/>
    <arg name="robot_name" default="Bobby"/>
    <arg name="controllers" default="true"/>
    <arg name="joy_device" default="/dev/input/js0"/>
    <arg name="joystick_connected" default="true"/>
    <arg name="autorespawn" default="true"/>
    <arg name="world_name" default="sandy_path.world"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find bobby)/worlds/surface_testing.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="recording" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!--Launch controllers-->
    <group if="$(arg controllers)">
        <include file="$(find bobby)/launch/controllers_tracked.launch">
            <arg name="rqt" value="$(arg rqt)"/>
            <arg name="robot_name" value="$(arg robot_name)"/>
            <arg name="controllers" value="$(arg controllers)"/>
            <arg name="joy_device" value="$(arg joy_device)"/>
            <arg name="joystick_connected" value="$(arg joystick_connected)"/>
            <arg name="autorespawn" value="$(arg autorespawn)"/>
        </include>

    </group>


    <group if="$(arg rqt)">
        <node name="rqt" pkg="rqt_gui" type="rqt_gui" respawn="$(arg autorespawn)"/>
    </group>

    <group if="$(arg joystick_connected)">
        <node ns="$(arg robot_name)" name="joy_listener" pkg="bobby" type="tracks_joy_listener.py"
           respawn="$(arg autorespawn)" output="screen"/>
        <node respawn="$(arg autorespawn)" pkg="joy"
              type="joy_node" name="joy" >
            <param name="dev" type="string" value="$(arg joy_device)"/>
            <param name="deadzone" value="0.1" />
        </node>
    </group>
</launch>
